scour/scour.py

```diff
@@ -3413,6 +3413,11 @@ def remapNamespacePrefix(node, oldprefix, newprefix):
 def make_well_formed(text, quote_dict=None):
     if quote_dict is None:
         quote_dict = XML_ENTS_NO_QUOTES
+    if not any(c in text for c in quote_dict):
+        # The quote-able characters are quite rare in SVG (they mostly only
+        # occur in text elements in practice).  Therefore it make sense to
+        # optimize for this common case
+        return text
     return ''.join(quote_dict[c] if c in quote_dict else c for c in text)
```

Comment:
It is a micro-optimization and I cannot think of anything else using pure python that would make the original statement go notably faster.  One of the wins here is to avoid allocate memory to clone a string when we can just return the original one.  Obviously, if we start to look at C or Cython, then our options expand (but so does the complexity of the optimization).

The only final option, I can think of (but did not test), was to use a cache mechanism to avoid triggering the case in the first place (a la `@lru_cache`). But if we go down that route, then `quote_dict` should probably no longer be an input parameter (to keep the caching simple and cheap).

It is relevant to know that we rarely ever need to use the escape/quote sequences in practise as most attributes have a well-defined set of values that can be inserted.

Reply:
Great, seems you already put a lot of thought into it, so I trust in your choice that this is the best way to go.